### PR TITLE
Feature/rest api

### PR DESCRIPTION
**Pull request by jdoppelreiter**

### CODE REVIEW

File: src/migrations/22.2_to_23.1/feature-rest-api.sql [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+, lgrt_error_code           number
+, lgrt_error_msg            VARCHAR2(2000 CHAR)
+, lgrt_error_stacktrace     CLOB
+);
+
+insert into flow_configuration(cfig_key, cfig_value) values('logging_rest_incoming_calls','Y');
+insert into flow_configuration(cfig_key, cfig_value) values('logging_rest_incoming_calls_retain_days','60');
+insert into flow_configuration(cfig_key, cfig_value) values('rest_base',null);
+
+create type flow_rest_role_ot as object
```
Comment (commi235) [inline src/migrations/22.2_to_23.1/feature-rest-api.sql:20]: Do we need Schema level types?

File: src/views/flow_rest_process_vars_vw.sql [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  as
+  select  pv.prov_prcs_id  as prcs_id
+        , pv.prov_scope    as scope
+        , pv.prov_var_name as name
+        , pv.prov_var_type as type
+        , decode( lower(pv.prov_var_type) 
+              , 'varchar2', pv.prov_var_vc2
+              , 'number',   pv.prov_var_num
+              , 'date',     pv.prov_var_num
+              , 'clob',     pv.prov_var_num
```
Comment (commi235) [inline src/views/flow_rest_process_vars_vw.sql:18]: Doesn't look right

File: src/types/flow_rest_role_ot.sql
Comment (commi235) [inline src/types/flow_rest_role_ot.sql:1]: Check if we need Schema level types

File: src/types/flow_rest_roles_nt.sql
Comment (commi235) [inline src/types/flow_rest_roles_nt.sql:1]: Schema level types continued

File: src/plsql/flow_rest_auth.pkb [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+    l_roles(1) := flow_rest_constants.c_rest_role_read;
+    l_roles(2) := flow_rest_constants.c_rest_role_write;
+    l_roles(3) := flow_rest_constants.c_rest_role_admin;
+
+    for rec_roles in (select id, name 
+                        from user_ords_roles 
+                        where name in (select column_value from table(l_roles)))
+    loop
+      PIPE ROW (flow_rest_role_ot ( rec_roles.id, rec_roles.name ));
```
Comment (commi235) [inline src/plsql/flow_rest_auth.pkb:19]: Pipelined table function does not need Schema level types.
Types can be declared in the package header..

**Review by commi235 — COMMENTED** [review]
I'd suggest to move the types into a package as it seems we don't need Schema level types.

**Review by commi235 — APPROVED** [review]

File: src/views/flow_rest_process_vars_vw.sql (hunk 2 above)
Comment (commi235) [reply src/views/flow_rest_process_vars_vw.sql:18]: date and clob shouldn't use pro_var_num here.

**Review by commi235 — CHANGES_REQUESTED** [review]
flow_rest_process_vars_vw doesn't look right.

**Comment by jdoppelreiter** [comment]
Thanks for finding it... view flow_rest_process_vars_vw adapted

**Review by commi235 — APPROVED** [review]

**Pull request merged**